apps/load-generator/main.py

```diff
@@ -78,13 +78,22 @@ def run(self):
                 time.sleep(wait)
 
     def query(self, q):
-        start = time.time()
-        resp = requests.get(self.url, params={"query": q})
-        
-        dur = time.time() - start
-        print("query %s %s, status=%s, size=%d, dur=%d" %(self.t, q, resp.status_code, len(resp.text), dur))
+        try:
+            start = time.time()
+
+            params={"query": q["queryString"]}
+            if q["type"] == "query_range":
+                params["start"] = q["start"]
+                params["end"] = q["end"]
+                params["step"] = q["step"]
+            resp = requests.get(self.url + q["type"] + "?", params)
+
+            dur = time.time() - start
+            print("query %s %s, status=%s, size=%d, dur=%d" %(self.t, q, resp.status_code, len(resp.text), dur))
 
-        self.query_time.labels(self.t, str(self.i), q).observe(dur)
+            self.query_time.labels(self.t, self.groupName, q).observe(dur)
+        except requests.exceptions.ConnectionError as e:
+            print("Could not query prometheus instance %s. \n %s" %(self.url, e))
```

Comment:
We should add two counter metrics here tracking a) total queries and b) failed queries.
The `except` should probably just catch everything. I can't really think of a scenario where we want to crash the thread.